01_fibonacci/fibi.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+//prints the fibonacci series
+func fibonacci(n int) {
+	a, b := 0, 1
+	if n > 0 {
+		fmt.Fprintln(out, a)
+	}
+	if n >= 1 {
+		fmt.Fprintln(out, b)
+	}
```

Comment:
You can eliminate the above tests and `fmt.Fprintln` calls if you carefully choose the initial loop conditions and change what you print inside the loop.

Reply:
Modified